src/NServiceBus.Core/Serializers/SimpleJson/SimpleJson.g.cs

```diff
@@ -1067,7 +1067,24 @@ static bool SerializeObject(IJsonSerializerStrategy jsonSerializerStrategy, IEnu
                 if (stringKey != null)
                     SerializeString(stringKey, builder);
                 else
-                    SerializeString(key.ToString(), builder);
+                {
+//                    alternative approach:
+//                    jsonSerializerStrategy.TrySerializeNonPrimitiveObject(key, out var keyValue);
+//                    SerializeString((keyValue as string) ?? key.ToString(), builder);
+
+                    switch (key)
```

Comment:
downside of this approach is they duplicate `DateTime` serialization logic and that it only handles `DateTimes`, nothing else (wouldn't profit from types additionally supported by `TrySerializeNonPrimiteObject`.